### PR TITLE
HP-1686 better delete profile error handling

### PR DESCRIPTION
**Pull request by NikoHelle**
Show list of services which prevent profile deletion.

### CODE REVIEW

File: src/i18n/en.json [hunk 1]
```diff
@@ -37,21 +37,23 @@
...
+    "deleteFailed": "The deletion failed for some reason. Please try again after a while!"
   },
   "deleteProfileModal": {
     "delete": "Delete my information",
     "explanation": "By continuing you will lose all of your stored data and you can no longer use the following City of Helsinki services:",
     "noServiceExplanation": "By continuing you will lose all of your stored data.",
     "title": "Are you sure you want to delete your information?",
+    "notAllowed": "You cannot delete your profile because one of the services using it needs your information, for example, for invoicing.",
+    "deletionErrorTitle": "The deletion failed!",
+    "genericError": "The profile could not be deleted. Please try again later.",
```
Comment (kanguran) [inline src/i18n/en.json]: deleteFailed and genericError are pretty similar Error messages, do we really need to duplicate them ?

File: src/profile/components/deleteProfile/__tests__/DeleteProfile.test.tsx [hunk 2]
```diff
@@ -174,4 +196,27 @@ describe('<DeleteProfile /> ', () => {
...
+      });
+    });
+  });
+  it(`When deletion returns a generic error, an error message is shown`, async () => {
+    await act(async () => {
+      const testTools = await initTests(1);
+      const { waitForElement } = testTools;
+      await proceedUIToDeletionConfimed(testTools);
+      await waitForElement(errorDescriptionSelector);
+      expect(mockHistoryPushListener).toHaveBeenCalledTimes(0);
```
Comment (kanguran) [inline src/profile/components/deleteProfile/__tests__/DeleteProfile.test.tsx:219]: Add waitFor as in other places

      await waitFor(() => {
        expect(mockHistoryPushListener).toHaveBeenCalledTimes(0);
      });

File: .env.development [hunk 3]
```diff
@@ -2,4 +2,5 @@ REACT_APP_OIDC_AUTHORITY="https://tunnistamo.dev.hel.ninja/"
 REACT_APP_OIDC_CLIENT_ID="https://api.hel.fi/auth/helsinkiprofile-ui-dev"
 REACT_APP_PROFILE_AUDIENCE="https://api.hel.fi/auth/helsinkiprofiledev"
 REACT_APP_PROFILE_GRAPHQL="https://profile-api.dev.hel.ninja/graphql/"
-REACT_APP_OIDC_SCOPE="openid profile https://api.hel.fi/auth/helsinkiprofiledev"
\ No newline at end of file
+REACT_APP_OIDC_SCOPE="openid profile https://api.hel.fi/auth/helsinkiprofiledev"
+REACT_APP_PROFILE_BE_GDPR_CLIENT_ID="https://api.hel.fi/auth/helsinkiprofiledev"
```
Comment (kanguran) [inline .env.development:6]: Can't we use "https://api.hel.fi/auth/helsinkiprofiledev" as parameter here.

Comment (NikoHelle) [reply .env.development:6]: What do you mean "as parameter"?

File: src/i18n/en.json (hunk 1 above)
Comment (NikoHelle) [reply src/i18n/en.json]: Good catch. Removed the old one (genericError)

File: src/profile/components/deleteProfile/__tests__/DeleteProfile.test.tsx (hunk 2 above)
Comment (NikoHelle) [reply src/profile/components/deleteProfile/__tests__/DeleteProfile.test.tsx:219]: No need in this case. WaitFor is necessary when the test won't resolve immediately and need to wait for component update.

File: .env.development (hunk 3 above)
Comment (NikoHelle) [reply .env.development:6]: *.env could use this, BUT it won't work in Azure, so it was removed earlier.

REACT_APP_PROFILE_BE_GDPR_CLIENT_ID="https://api.hel.fi/auth/helsinkiprofiledev"
REACT_APP_OIDC_SCOPE="openid profile ${REACT_APP_PROFILE_BE_GDPR_CLIENT_ID}"

Comment (kanguran) [reply .env.development:6]: ok, clear

**Review by kanguran — APPROVED** [review]
Looks good now

**Pull request merged**